### PR TITLE
Fix automod url regex and logic

### PR DESCRIPTION
**Pull request by Zickles**
## Changes

<details>
<summary>Screenshots</summary>

</details>

<details>
<summary>Checkboxes</summary>

- [ ] I've added new features.
- [x] I've fixed bug. (_optional_ you can mention a issue if there is one)
- [ ] I've corrected the spelling in README, documentation, etc.
- [ ] I've check for issues. (`npm run lint`)
- [ ] I've fixed my formatting. (`npm run prettier`)

</details>

### CODE REVIEW

File: a.ts [hunk 1]
```diff
@@ -1,36 +1,31 @@
 const urlRegex =
```
Comment (Kathund) [inline a.ts]: Remove a.ts testing file

File: src/events/messageCreate.ts [hunk 2]
```diff
@@ -1,32 +1,33 @@
...
+  const domain: string = match[3];
+
+  if (allowedDomains.some((pattern) => pattern === domain) && !match[2]) {
+    return true;
+  } else if (!allowedDomains.some((pattern) => pattern === domain)) {
+    return (
+      allowedDomains.some((pattern) => pattern === `*.${domain}`) ||
+      allowedDomains.some((pattern) => pattern === match[2] + domain)
+    );
+  } else {
```
Comment (Kathund) [inline src/events/messageCreate.ts]: Check warning on line 28 in src/events/messageCreate.ts

GitHub Actions
/ check linting (eslint)

Unnecessary 'else' after 'return'

**Review by Kathund — CHANGES_REQUESTED** [review]

File: src/events/messageCreate.ts (hunk 2 above)
Comment (Zickles) [reply src/events/messageCreate.ts]: Lint may say unnecessary but typescript gives error bc there is no default return if you remove the else

Comment (Zickles) [reply src/events/messageCreate.ts]: nvm lmao

**Review by Kathund — APPROVED** [review]

**Pull request merged**